go/store/nbs/store.go

```diff
@@ -1655,6 +1688,8 @@ func (nbs *NomsBlockStore) swapTables(ctx context.Context, specs []tableSpec) (e
 		return errors.New("concurrent manifest edit during GC, before swapTables. GC failed.")
 	}
 
+	nbs.hasCache.Purge()
```

Comment:
Not clear from casual read why this is necessary here and only here

Reply:
Good point. Will add a comment.